jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/ReadPostsControllerTest.java

```diff
@@ -80,4 +85,12 @@ public void markAllTopicsAsReadShouldMarkThemAndUpdatePage() throws NotFoundExce
         assertEquals(result, "redirect:/branches/" + String.valueOf(markedBranchId));
         verify(lastReadPostService).markAllTopicsAsRead(willBeMarkedBranch);
     }
+
+    @Test
+    public void markTopicAsReadByIdShouldMarkItAsRead() throws NotFoundException {
```

Comment:
Not sure if this test is useful - we can simply write one component test that checks the result along with the DB calls instead of mocking all the DOCs.